src/execution/physical_plan/plan_aggregate.cpp

```diff
@@ -174,24 +174,38 @@ PhysicalPlanGenerator::ExtractAggregateExpressions(unique_ptr<PhysicalOperator>
 	vector<unique_ptr<Expression>> expressions;
 	vector<LogicalType> types;
 
-	for (idx_t group_idx = 0; group_idx < groups.size(); group_idx++) {
-		auto &group = groups[group_idx];
+	for (auto &group : groups) {
 		auto ref = make_unique<BoundReferenceExpression>(group->return_type, expressions.size());
 		types.push_back(group->return_type);
 		expressions.push_back(move(group));
-		groups[group_idx] = move(ref);
+		group = move(ref);
 	}
 
 	for (auto &aggr : aggregates) {
 		auto &bound_aggr = (BoundAggregateExpression &)*aggr;
-		for (idx_t child_idx = 0; child_idx < bound_aggr.children.size(); child_idx++) {
-			auto &child = bound_aggr.children[child_idx];
-			auto ref = make_unique<BoundReferenceExpression>(child->return_type, expressions.size());
-			types.push_back(child->return_type);
-			expressions.push_back(move(child));
-			bound_aggr.children[child_idx] = move(ref);
+		for (auto &child_ : bound_aggr.children) {
+			bool already_in = false;
+			for (size_t i = 0; i < expressions.size(); i++) {
+				auto *base_expr = (BaseExpression *)expressions[i].get();
+				if (child_->Equals(base_expr)) {
```

Comment:
Is this necessary for correctness purposes; or just an optimization? Not that I disagree with adding this, just for clarification. I would like to move it to a function, use an `expression_map_t` instead of a vector, and also use it for the `bound_aggr.filter`.